src/dist/manifestation/tests.rs

```diff
@@ -535,195 +400,205 @@ impl Compressions {
     }
 }
 
-fn setup(
-    edit: Option<&dyn Fn(&str, &mut MockChannel)>,
-    comps: Compressions,
-    f: &dyn Fn(&Url, &ToolchainDesc, &InstallPrefix, &DownloadCfg<'_>, &temp::Context),
-) {
-    let dist_tempdir = tempfile::Builder::new().prefix("rustup").tempdir().unwrap();
-    let mock_dist_server = create_mock_dist_server(dist_tempdir.path(), edit);
-    let url = Url::parse(&format!("file://{}", dist_tempdir.path().to_string_lossy())).unwrap();
-    setup_from_dist_server(mock_dist_server, &url, comps, f);
+struct TestContext {
+    url: Url,
+    toolchain: ToolchainDesc,
+    prefix: InstallPrefix,
+    download_dir: PathBuf,
+    tp: TestProcess,
+    tmp_cx: temp::Context,
+    _tempdirs: Vec<tempfile::TempDir>,
 }
 
-fn setup_from_dist_server(
-    server: MockDistServer,
-    url: &Url,
-    comps: Compressions,
-    f: &dyn Fn(&Url, &ToolchainDesc, &InstallPrefix, &DownloadCfg<'_>, &temp::Context),
-) {
-    server.write(
-        &[MockManifestVersion::V2],
-        comps.enable_xz(),
-        comps.enable_zst(),
-    );
+impl TestContext {
+    fn new(edit: Option<&dyn Fn(&str, &mut MockChannel)>, comps: Compressions) -> Self {
```

Comment:
I think this could benefit from a more descriptive name. I feel like now the relation between `new()` and `from_dist_server()` is the opposite from what I expect (`new()` is generally lower level than other constructors, but in this case it is arguably higher level.)

Reply:
@djc Hmmm, `setup` used to be a short name and I changed it to `new` simply because of the "more uses = terser name" convention.

That said, what about `Self::from_edit()`? I'm really bad at naming 🤣

Reply:
Hmm, on review I feel like `new()` is probably reasonable here based on how it is used, even if it is a little counterintuitive to me. Let's just merge it?